joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -93,29 +93,47 @@ Trajectory::sample(
     return false;
   }
 
-  auto deduce_from_derivaties = [&](trajectory_msgs::msg::JointTrajectoryPoint & first_state,
+  auto integrate_from_derivatives = [&](trajectory_msgs::msg::JointTrajectoryPoint & first_state,
       trajectory_msgs::msg::JointTrajectoryPoint & second_state,
       const size_t dim, const double delta_t)
     {
-      if (second_state.positions.empty()) {
-        second_state.positions.resize(dim);
-        if (first_state.velocities.empty()) {
-          first_state.velocities.resize(dim, 0.0);
-        }
-        if (second_state.velocities.empty()) {
-          second_state.velocities.resize(dim);
-          if (first_state.accelerations.empty()) {
-            first_state.accelerations.resize(dim, 0.0);
+      // Previous position must be known already
+      if (first_state.positions.empty()) {
+        throw std::runtime_error("Integration failed. A required position term is unknown.");
+      }
+      // We can skip these terms, e.g. if second_state.positions or .accelerations is missing
+      // Use references for readability
+      const auto & have_first_velocities = !first_state.velocities.empty();
+      const auto & have_first_accelerations = !first_state.accelerations.empty();
+      const auto & have_second_accelerations = !second_state.accelerations.empty();
```

Comment:
Not sure we need "&" because the type is bool.

```suggestion
      const auto have_first_velocities = !first_state.velocities.empty();
      const auto have_first_accelerations = !first_state.accelerations.empty();
      const auto have_second_accelerations = !second_state.accelerations.empty();
```

Reply:
My question here is:

`const auto have_first_velocities = !first_state.velocities.empty();`

Will allocate a new variable and not be realtime safe, right?